radixdlt-java/src/main/java/com/radixdlt/client/core/ledger/RadixClientSupplier.java

```diff
@@ -0,0 +1,177 @@
+
+	/**
+	 * Create a client selector from a certain network with the default filters and the a randomized selector
+	 * @param network The network
+	 */
+	public RadixClientSupplier(RadixNetwork network) {
+		this(network, new RandomSelector(), Arrays.asList(
+				new ConnectionAliveFilter(),
+				new CompatibleApiVersionFilter(RadixJsonRpcClient.API_VERSION)));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a matching universe filter, the default filters and a randomized selector
+	 * @param network The network
+	 * @param config The universe config for the matching universe filter
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixUniverseConfig config) {
+		this(network, new RandomSelector(), Arrays.asList(
+				new ConnectionAliveFilter(),
+				new CompatibleApiVersionFilter(RadixJsonRpcClient.API_VERSION),
+				new MatchingUniverseFilter(config)));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a certain selector and filters
+	 * @param network The network
+	 * @param selector The selector used to select a peer from the desirable peer list
+	 * @param filters The filters used to test the desirability of peers
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixPeerSelector selector, RadixPeerFilter... filters) {
+		this(network, selector, Arrays.asList(filters));
+	}
+
+	/**
+	 * Create a client selector from a certain network with a certain selector and filters
+	 * @param network The network
+	 * @param selector The selector used to select a peer from the desirable peer list
+	 * @param filters The filters used to test the desirability of peers
+	 */
+	public RadixClientSupplier(RadixNetwork network, RadixPeerSelector selector, List<RadixPeerFilter> filters) {
+		Objects.requireNonNull(network, "network is required");
+		Objects.requireNonNull(selector, "selector is required");
+		Objects.requireNonNull(filters, "filters is required");
+
+		this.network = network;
+		this.selector = selector;
+		this.filters = filters;
+		this.delaySecs = 3;
+	}
+
+	/**
+	 * Returns a cold single of the first peer found which supports
+	 * a set short shards which intersects with a given shard
+	 *
+	 * @param shard a shards to find an intersection with
+	 * @return a cold single of the first matching Radix client
+	 */
+	public Single<RadixJsonRpcClient> getRadixClient(Long shard) {
+		return getRadixClient(Collections.singleton(shard));
+	}
+
+	/**
+	 * Returns a cold single of the first peer found which supports
+	 * a set short shards which intersects with a given set of shards.
+	 *
+	 * @param shards set of shards to find an intersection with
+	 * @return a cold single of the first matching Radix client
+	 */
+	public Single<RadixJsonRpcClient> getRadixClient(Set<Long> shards) {
+		if (shards.isEmpty()) {
+			throw new IllegalArgumentException("Shards cannot be empty to obtain a radixClient.");
+		}
+
+		List<RadixPeerFilter> expandedFilters = new ArrayList<>(this.filters);
```

Comment:
`this.filters` needs to be thread protected or become an unmodifiableList (probably the latter).